crates/c-api/src/extern.rs

```diff
@@ -20,7 +20,7 @@ pub extern "C" fn wasm_extern_kind(e: &wasm_extern_t) -> wasm_externkind_t {
         Extern::Global(_) => crate::WASM_EXTERN_GLOBAL,
         Extern::Table(_) => crate::WASM_EXTERN_TABLE,
         Extern::Memory(_) => crate::WASM_EXTERN_MEMORY,
-        Extern::SharedMemory(_) => todo!(),
+        Extern::SharedMemory(_) => crate::WASM_EXTERN_SHAREDMEMORY,
```

Comment:
There's a split between the `wasm_*` APIs and `wasmtime_*` APIs and we can't change the `wasm_*` ones so this unfortunately can't change. Perhaps rename the `todo!()` though to a `panic!()` with an error message along the lines of "please use the `wasmtime_*` method instead"?

I should also note that if the `wasmtime_*` method doesn't exist yet and you need this method specifically then it's fine to add a `wasmtime_*` method too.